services/processing/merging.py

```diff
@@ -34,9 +34,14 @@ def merge_reports(
 
         old_sessionid = next(iter(report.sessions))
         new_sessionid = master_report.next_session_number()
-        change_sessionid(report, old_sessionid, new_sessionid)
         session_mapping[intermediate_report.upload_id] = new_sessionid
 
+        if master_report.is_empty() and old_sessionid == new_sessionid:
+            # if the master report is empty, we can avoid a costly merge operation
```

Comment:
This happens the very first time there's an upload for a commit right?

Reply:
yes exactly, except when using carry-forwards